reference/halyard/high-availability.md

```diff
@@ -0,0 +1,115 @@
+
+__Important:__ Please note that Halyard only supports this functionality for a [distributed Spinnaker deployment](/setup/install/environment/#distributed-installation) configured with a [manifest based Kubernetes account](/setup/install/providers/kubernetes-v2/).
+
+## HA Clouddriver
+
+Clouddriver benefits greatly from isolating its operations into separate services. To split Clouddriver for increased availability, run:
+
+```hal config deploy ha clouddriver enable```
+
+When Spinnaker is deployed with this flag enabled, Clouddriver will be deploy as three different services, each only performing a subset of the base Clouddriver's operations. Although by default the three Clouddriver services will communicate with the global Redis (all Spinnaker services speak to this Redis) provided by Halyard, it is recommended that the logical Clouddriver services be configured to communicate with an external Redis service. To be most effective, the Clouddriver-Read-Only should be configured to speak to a Redis slave, while the other two should be configured to speak to the master. This is handled by automatically by Halyard if the user provides the two endpoints using this command:
+
+```hal config deploy ha services clouddriver edit --redis-master-endpoint $REDIS_MASTER_ENDPOINT --redis-slave-endpoint $REDIS_SLAVE_ENDPOINT```
+
+### Clouddriver-Caching
+
+The first of the three logical Clouddriver services is the Clouddriver-Caching service, or ```clouddriver-caching```. This service is responsible for the caching and retrieving of cloud infrastructure data. Since this is all that the Clouddriver-Caching is doing, there is no communication between this service and any other Spinnaker service.
+
+This service's name when [configuring its sizing](/reference/halyard/component-sizing/) is ```spin-clouddriver-caching```.
+
+To add a [custom profile](/reference/halyard/custom/#custom-profiles) or [custom service settings](/reference/halyard/custom/#custom-service-settings) for this service, use the name ```clouddriver-caching```.
+
+### Clouddriver-Read-Write
+
+The second logical Clouddriver service is the Clouddriver-Read-Write service, or ```clouddriver-rw```. This service handles all mutation operations aside from what the Clouddriver-Caching service does. This service can be scaled to handle an increased number of writes.
+
+This service's name when [configuring its sizing](/reference/halyard/component-sizing/) is ```spin-clouddriver-rw```.
+
+To add a [custom profile](/reference/halyard/custom/#custom-profiles) or [custom service settings](/reference/halyard/custom/#custom-service-settings) for this service, use the name ```clouddriver-rw```.
+
+### Clouddriver-Read-Only
+
+Inspired by [Netflix](https://blog.spinnaker.io/scaling-spinnaker-at-netflix-part-1-8a5ae51ee6de), the Clouddriver-Read-Only service, or ```clouddriver-ro``` handles all read requests to Clouddriver. This service can be scaled to handle an increased number of reads.
```

Comment:
I think each of these configurations come from Netflix's large scale experience described in that doc. I'd either mention it more holistically further up in the doc or I'd exclude it entirely.